pymodbus/transport/transport.py

```diff
@@ -156,17 +156,21 @@ def __init__(
         self.is_server = is_server
         self.is_closing = False
 
-        self.transport: asyncio.BaseTransport = None
-        self.loop: asyncio.AbstractEventLoop = None
+        self.loop: asyncio.AbstractEventLoop = None  # type: ignore[assignment]
+        self.transport: asyncio.BaseTransport | None = None
         self.recv_buffer: bytes = b""
-        self.call_create: Callable[[], Coroutine[Any, Any, Any]] = lambda: None
+
+        async def _noop():
+            ...
+
+        self.call_create: Callable[[], Coroutine[Any, Any, Any]] = _noop
         if self.is_server:
             self.active_connections: dict[str, ModbusProtocol] = {}
         else:
-            self.listener: ModbusProtocol = None
+            self.listener: ModbusProtocol | None = None
             self.unique_id: str = str(id(self))
-            self.reconnect_task: asyncio.Task = None
-            self.reconnect_delay_current: float = 0.0
+            self.reconnect_task: asyncio.Task | None = None
+            self.reconnect_delay_current: float | None = 0.0
```

Comment:
this cannot be None as far as I remember the code... 0.0 is not None.

Reply:
It's unclear what the intended value of `self.reconnect_delay_current` is when `self.reconnect_delay == None`.

I could not easily decide between `0.0` and `None` in that case, so I went with `None` for fewer code changes.

Reply:
self.reconnect_delay_current is a value between self.reconnect_delay and self.reconnect_delay_max. so setting it to self.reconnect_delay is probably the most generic way.